### ISSUE TITLE
With Node 6, Gulp outputs more JS stack traces than useful messages

### ISSUE THREAD

**Issue by JohnAlbin** [gulpjs/gulp#1628]
I tried out Node 6 and, after deleting node_modules and running `npm install`, when I run `gulp`, I receive pages and pages of "JS stack trace"s. Starting with:

> (node:94076) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.

Here's the full output: http://pastebin.com/6Zxgz8Ar

It makes it really, _really_ difficult to use Gulp 3 on Node 6 if I have difficulty finding the normal output amongst the gobs of undesirable stack traces.

I'm fairly certain this is the fall out of #1571 

I know that newer vinyl-fs versions don't use graceful-fs 3 or earlier, but we can't upgrade to that because it will cause API breakage.

One solution to this would be to update an older version vinyl-fs to not use graceful-fs 3 or earlier. And then to update gulp 3.9.x branch with the fixed dependencies.

Given the number of issues I see open for vinyl-fs, it doesn't look like Gulp 4 will be released quick enough to resolve this Node 6 issue in a timely matter (1-2 weeks from now.) Homebrew just upgraded me to Node 6 and the number of people who are upgrading is going to keep growing, so this is a critical issue, imo.

**Comment by JohnAlbin** [comment]
BTW, this is how you downgrade to Node 5 in homebrew:

```
brew uninstall node
brew tap homebrew/versions
brew install node5
```

**Comment by mgol** [comment]
FWIW, I no longer see the long stack traces with gulp, just a warning:
```
(node:31726) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
(node:31726) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
```
and the rest of the output is normal so this doesn't seem so severe now. Maybe sth changed in the logging level? Could you remove `node_modules,` install packages again (making sure you use the latest version of everything) and check again?

**Comment by swernerx** [comment]
It's actually the same issue. Even when deleting the package folder and reinstalling everything.

It seems to be related to the old version of "graceful-fs"

If I do a "npm ls graceful-fs" - as suggested in the debug output - I see the following in my project:

```
⚡ npm ls graceful-fs
modern-spa-boilerplate@0.1.1 /Users/Sebastian/Workspace/sebastian-software/modern-spa-boilerplate
├─┬ ava@0.14.0
│ └─┬ chokidar@1.4.3
│   └─┬ fsevents@1.0.12
│     └─┬ node-pre-gyp@0.6.25
│       └─┬ tar@2.2.1
│         └─┬ fstream@1.0.8
│           └── graceful-fs@4.1.3
├─┬ gulp@3.9.1
│ └─┬ vinyl-fs@0.3.14
│   ├─┬ glob-watcher@0.0.6
│   │ └─┬ gaze@0.5.2
│   │   └─┬ globule@0.1.0
│   │     └─┬ glob@3.1.21
│   │       └── graceful-fs@1.2.3
│   └── graceful-fs@3.0.8
└─┬ gulp-sourcemaps@2.0.0-alpha
  └── graceful-fs@4.1.3
```

The only old version < 4 is used by the "glob" package. Maybe this can be fixed - as it's pretty darn old.

Easiest upgrade would be possibly to update to a more recent version of "glob-watcher" which is currently at 3.0.0.

**Comment by mgol** [comment]
@JohnAlbin I've looked at your gist and it's

**Comment by mgol** [comment]
@JohnAlbin I've looked at your gist and it's node-sass that generates those stack traces, not Gulp. Gulp only prints a small harmless warning and that won't be fixed in Gulp v3 according to #1571. The node-sass issue is https://github.com/sass/node-sass/issues/1484, it's already fixed and the next node-sass version will contain the fix.

So I think this issue should be closed as it's unrelated to Gulp.

**Comment by phated** [comment]
@mgol thanks for looking into this and giving a clear response!
